pkg/handlers/deploy.go

```diff
@@ -140,5 +166,144 @@ func (h *Handler) DeployAppVersion(w http.ResponseWriter, r *http.Request) {
 
 	deployAppVersionResponse.Success = true
 
-	JSON(w, 204, deployAppVersionResponse)
+	JSON(w, http.StatusOK, deployAppVersionResponse)
+}
+
+func isKotsUpgradeNeeded(app *apptypes.App, sequence int64) (bool, string, error) {
+	archivePath, err := ioutil.TempDir("", "kotsadm")
+	if err != nil {
+		return false, "", errors.Wrap(err, "failed to create temp dir")
+	}
+	defer os.RemoveAll(archivePath)
+
+	err = store.GetStore().GetAppVersionArchive(app.ID, int64(sequence), archivePath)
+	if err != nil {
+		return false, "", errors.Wrapf(err, "failed to get archive for sequence %d", sequence)
+	}
+
+	kotsKinds, err := kotsutil.LoadKotsKindsFromPath(archivePath)
+	if err != nil {
+		return false, "", errors.Wrap(err, "failed to load kots kinds")
+	}
+
+	latestVersion, _ := findLatestKotsVersion(app.ID, kotsKinds.License)
+
+	targetVersion, err := getTargetKotsVersion(kotsKinds, latestVersion)
+	if err != nil {
+		if err, ok := err.(AdminConsoleUpgradeError); ok {
+			if err.IsCritical {
+				return false, "", errors.Wrap(err, "critical error while checking target kots version")
+			}
+			logger.Infof("skipping kots upgrade because %v", err)
+			return false, "", nil
+		}
+		return false, "", errors.Wrap(err, "failed to find target version")
+	}
+
+	return true, targetVersion, nil
+}
+
+func findLatestKotsVersion(appID string, license *kotsv1beta1.License) (string, error) {
```

Comment:
what do you think about moving this function to the `kotsutil` package? like `pkg/kotsutil/kots.go`

Reply:
This function has `reporting` as a dependency.  Also I kinda have the kotsutil package :)